client/installation-helper.cc

```diff
@@ -0,0 +1,226 @@
+
+    cout << "done" << endl;
+}
+
+
+void
+step2(const string& device, const string& root_prefix, const string& default_subvolume_name)
+{
+    // step runs in inst-sys
+
+    cout << "step 2 device:" << device << " root-prefix:" << root_prefix
+	 << " default-subvolume-name:" << default_subvolume_name << endl;
+
+    cout << "mounting device" << endl;
+
+    string subvol_option = default_subvolume_name;
+    if (!subvol_option.empty())
+	subvol_option += "/";
+    subvol_option += ".snapshots";
+
+    SDir s_dir(root_prefix + "/.snapshots");
+    if (!s_dir.mount(device, "btrfs", 0, "subvol=" + subvol_option))
+    {
+	cerr << "mounting .snapshots failed" << endl;
+    }
+
+    cout << "done" << endl;
+}
+
+
+void
+step3(const string& root_prefix, const string& default_subvolume_name)
+{
+    // step runs in inst-sys
+
+    cout << "step 3 root-prefix:" << root_prefix << " default_subvolume_name:"
+	 << default_subvolume_name << endl;
+
+    cout << "adding .snapshots to fstab" << endl;
+
+    Btrfs btrfs("/", root_prefix);
+
+    btrfs.addToFstab(default_subvolume_name);
+
+    cout << "done" << endl;
+}
+
+
+void
+step4()
+{
+    // step runs in chroot
+
+    cout << "step 4" << endl;
+
+    cout << "modifying sysconfig-file" << endl;
+
+    try
+    {
+	SysconfigFile sysconfig(SYSCONFIGFILE);
+	sysconfig.setValue("SNAPPER_CONFIGS", { "root" });
+    }
+    catch (const FileNotFoundException& e)
+    {
+	cerr << "sysconfig-file not found" << endl;
+    }
+
+    Btrfs btrfs("/", "");
+
+    cout << "running external programs" << endl;
+
+    Hooks::create_config("/", &btrfs);
+
+    cout << "done" << endl;
+}
+
+
+int
+main(int argc, char** argv)
+{
+    setlocale(LC_ALL, "");
+
+    const struct option options[] = {
+	{ "step",			required_argument,	0,	0 },
+	{ "device",			required_argument,	0,	0 },
+	{ "root-prefix",		required_argument,	0,	0 },
+	{ "default-subvolume-name",	required_argument,	0,	0 },
+	{ "description",		required_argument,	0,	0 },
+	{ 0, 0, 0, 0 }
+    };
+
+    string step;
+    string device;
+    string root_prefix = "/";
+    string default_subvolume_name;
+    string description;
+
+    GetOpts getopts;
+
+    getopts.init(argc, argv);
+
+    GetOpts::parsed_opts opts = getopts.parse(options);
+
+    GetOpts::parsed_opts::const_iterator opt;
+
+    if ((opt = opts.find("step")) != opts.end())
+	step = opt->second;
+
+    if ((opt = opts.find("device")) != opts.end())
+	device = opt->second;
+
+    if ((opt = opts.find("root-prefix")) != opts.end())
+	root_prefix = opt->second;
+
+    if ((opt = opts.find("default-subvolume-name")) != opts.end())
+	default_subvolume_name = opt->second;
+
+    if ((opt = opts.find("description")) != opts.end())
+	description = opt->second;
+
+    if (step == "1")
+	step1(device, description);
+    else if (step == "2")
+	step2(device, root_prefix, default_subvolume_name);
+    else if (step == "3")
+	step3(root_prefix, default_subvolume_name);
+    else if (step == "4")
+	step4();
```

Comment:
Yes, internal tool only called from YaST. That's why it is placed in /usr/lib/snapper/...